### ISSUE TITLE
tasks: error message with parallel tasks is delayed until other task finishes

### ISSUE THREAD

**Issue by llacom** [babashka/babashka#847]
babashka v0.4.1
Mac M1 Big Sur 11.2.3

When running 2 tasks in parallel, if one crashes the error message won't be shown until the other task finishes.

This can be an issue specially if you're running two long-lived process (clojure + postcss for example), where one of them could be failling silently.

Example:

```
{:tasks
 a    (run '-a {:parallel true})
 -a  {:depends [a:a a:b]}
 a:a (shell "cat")
 a:b (shell "unknownbinary")
 }
```

Output:

```
`--> bb a
asd
asd
----- Error --------------------------------------------------------------------
Type:     java.util.concurrent.ExecutionException
Message:  java.io.IOException: Cannot run program "unknownbinary": error=2, No such file or directory
```

Running `bb a` will not show the error from `a:b` until you quit `cat` (through `Ctrl-D`, `Ctrl-C` will not show the error at all).

Changing the order (`:depends [a:b a:a]`) shows the error immediately then aborts.

I would expect the error message to appear immediately, then proceed (or abort?) with the other tasks.

**Comment by borkdude** [comment]
You're right, this should be improved

**Comment by borkdude** [comment]
Note to self: https://stackoverflow.com/a/19348417/6264

**Comment by borkdude** [comment]
@llacom Could you try with this new binary:

https://18981-201467090-gh.circle-artifacts.com/0/release/babashka-0.4.2-SNAPSHOT-macos-amd64.tar.gz

**Comment by llacom** [comment]
It works well! All tasks abort and an error message is written as soon as there's an exception thrown in one of the tasks.

**Comment by borkdude** [comment]
Released in 0.4.3
